mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -278,16 +204,31 @@ impl mayastor_server::Mayastor for MayastorSvc {
     async fn add_child_nexus(
         &self,
         request: Request<AddChildNexusRequest>,
-    ) -> GrpcResult<Null> {
+    ) -> GrpcResult<Child> {
         let args = request.into_inner();
         trace!("{:?}", args);
         let uuid = args.uuid.clone();
         debug!("Adding child {} to nexus {} ...", args.uri, uuid);
-        locally! { async move {
-            nexus_lookup(&args.uuid)?.add_child(&args.uri, args.norebuild).await.map(|_| ())
+        // XXX How to avoid following spaghetti code if we cannot
+        // use '?' inside 'locally' macro?
+        let child = locally! { async move {
```

Comment:
What if you put what's inside the locally into a function and then you just call it, would that work?

Reply:
yes, I could do that. I can create a special function in nexus_grpc.rs and call it from here. Not the best solution but likely better than this code.